pkg/eval/ievaluator.go

```diff
@@ -0,0 +1,14 @@
+package eval
+
+/*
+IEvaluator implementations store the state of the flags, do parsing and validation of the flag state and evaluate flags in response to handlers.
+*/
+type IEvaluator interface {
+	// TODO: add context param when rule evaluator is implemented
+	GetState() (string, error)
+	SetState(state string) error
```

Comment:
With this abstraction, it's actually possible to add IService impls that don't have well-defined transports as well... you could simply use GetState() to return the entire flag data, and only call that in whatever handler you implement.